client/client.go

```diff
@@ -442,9 +443,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		return nil, nil, err
 	}
 
+	isWait := false
+	if req != nil && req.URL != nil {
+		isWait, _ = strconv.ParseBool(req.URL.Query().Get("wait"))
+	}
+
 	var hctx context.Context
 	var hcancel context.CancelFunc
-	if c.headerTimeout > 0 {
+	if !isWait && c.headerTimeout > 0 {
```

Comment:
@gyuho 

This does not seem to be a 100% correct change. Even for the watch case, the header should be returned first before the body which is to wait on. I suspect that our etcd proxy does not forward header in time though. Not sure about other cases of proxy. We have to make a tradeoff here.